src/plugins/data/common/field_formats/converters/date_nanos.ts

```diff
@@ -70,65 +68,63 @@ export function formatWithNanos(
   }
 }
 
-export function createDateNanosFormat() {
-  return class DateNanosFormat extends FieldFormat {
-    static id = 'date_nanos';
-    static title = 'Date Nanos';
-    static fieldType = KBN_FIELD_TYPES.DATE;
+export class DateNanosFormat extends FieldFormat {
+  static id = 'date_nanos';
+  static title = 'Date Nanos';
+  static fieldType = KBN_FIELD_TYPES.DATE;
 
-    private getConfig: Function;
-    private memoizedConverter: Function = noop;
-    private memoizedPattern: string = '';
-    private timeZone: string = '';
+  private getConfig: Function;
+  private memoizedConverter: Function = noop;
+  private memoizedPattern: string = '';
+  private timeZone: string = '';
 
-    constructor(params: Record<string, any>, getConfig: Function) {
-      super(params);
+  constructor(params: Record<string, any>, getConfig: Function) {
+    super(params);
 
-      this.getConfig = getConfig;
-    }
+    this.getConfig = getConfig;
```

Comment:
It looks like the problem is in the constructor of the FieldFormat class. Please add getConfig method definition there